modules/operator/main.tf

```diff
@@ -0,0 +1,188 @@
+              iam = {
+                roles = {
+                  environment = var.iam_role_arn
+                }
+              }
+            }
+          }
+        }
+      }
+    })
+  ]
+
+  depends_on = [kubernetes_namespace.materialize]
+}
+
+resource "kubernetes_secret" "materialize_backends" {
+  for_each = { for idx, instance in var.instances : instance.name => instance }
+
+  metadata {
+    name      = "${each.key}-materialize-backend"
+    namespace = coalesce(each.value.namespace, var.operator_namespace)
+  }
+
+  data = {
+    metadata_backend_url = format(
+      "postgres://%s:%s@%s/%s?sslmode=require",
+      each.value.database_username,
+      each.value.database_password,
+      each.value.database_host,
+      coalesce(each.value.database_name, "${each.key}_db")
+    )
+    persist_backend_url = format(
+      "s3://%s/%s-%s:serviceaccount:%s:%s",
+      var.s3_bucket_name,
+      var.environment,
+      each.key,
+      coalesce(each.value.namespace, kubernetes_namespace.materialize.metadata[0].name),
+      each.value.instance_id
+    )
+  }
+}
+
+resource "kubernetes_manifest" "materialize_instances" {
+  for_each = { for idx, instance in var.instances : instance.name => instance }
+
+  manifest = {
+    apiVersion = "materialize.cloud/v1alpha1"
+    kind       = "Materialize"
+    metadata = {
+      name      = each.value.instance_id
+      namespace = coalesce(each.value.namespace, var.operator_namespace)
+    }
+    spec = {
+      environmentdImageRef = "materialize/environmentd:${var.operator_version}"
+      backendSecretName    = "${each.key}-materialize-backend"
+      environmentdResourceRequirements = {
+        limits = {
+          memory = each.value.memory_limit
+        }
+        requests = {
+          cpu    = each.value.cpu_request
+          memory = each.value.memory_request
+        }
+      }
+      balancerdResourceRequirements = {
+        limits = {
+          memory = "256Mi"
+        }
+        requests = {
+          cpu    = "100m"
+          memory = "256Mi"
+        }
+      }
+    }
+  }
+
+  depends_on = [
+    helm_release.materialize_operator,
+    kubernetes_secret.materialize_backends,
+    kubernetes_namespace.instance_namespaces
+  ]
+}
+
+# Materialize does not currently create databases within the instances, so we need to create them ourselves
+resource "kubernetes_manifest" "db_init_configmap" {
+  for_each = { for idx, instance in var.instances : instance.name => instance }
+
+  manifest = {
+    apiVersion = "v1"
+    kind       = "ConfigMap"
+    metadata = {
+      name      = "init-db-script-${each.key}"
+      namespace = coalesce(each.value.namespace, var.operator_namespace)
+    }
+    data = {
+      "init.sql" = format(
+        "CREATE DATABASE IF NOT EXISTS %s;",
+        coalesce(each.value.database_name, "${each.key}_db")
+      )
+    }
+  }
+}
+
+resource "kubernetes_manifest" "db_init_job" {
```

Comment:
if it works it works, but I'm a bit surprised we need to do this versus using the RDS provider to initialize the databases for us. this is more portable though!

Reply:
Yes, it would have been much more convenient, but it is not supported with the RDS terraform module. 

You can only initialize a single db:

https://registry.terraform.io/modules/terraform-aws-modules/rds/aws/latest#input_db_name